### ISSUE TITLE
waiting on create/update future for SQL Server - Code="UpsertLogicalServerRequestAlreadyInProgress" Message="An ongoing logical server request is already in progress, please try your request again later.

### ISSUE THREAD

**Issue by LaurentLesle** [hashicorp/terraform-provider-azurerm#12964]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* azurerm_mssql_server

### Terraform Configuration Files

https://github.com/aztfmod/terraform-azurerm-caf/blob/master/examples/mssql_server/105-sqlserver-failover_groups/configuration.tfvars

### Expected Behaviour

The provider should re-try until reaching the timeout

### Actual Behaviour

The CI is trying to configure a sql server failover group and the code fails after 1m50s with the error message:

module.example.module.mssql_servers["sql-server-rg1"].azurerm_mssql_server.mssql: Creation complete after 1m57s [id=/subscriptions/***/resourceGroups/sbkv-rg-sql-rg1/providers/Microsoft.Sql/servers/sbkv-sql-sql-server-rg1]

Error: waiting on create/update future for SQL Server "sbkv-sql-sql-server-rg2" (Resource Group "sbkv-rg-sql-rg2"): Code="UpsertLogicalServerRequestAlreadyInProgress" Message="An ongoing logical server request is already in progress, please try your request again later. (Server Name: sbkv-sql-sql-server-rg2, RequestId: fe3b2119-89ca-4d51-912e-fa4f7376ce7b, SubscriptionId: ***, ResourceGroup: sbkv-rg-sql-rg2)"

### Steps to Reproduce

https://github.com/aztfmod/terraform-azurerm-caf/runs/3307734102?check_suite_focus=true#step:5:748

### References

* #0000

**Comment by sicil1ano** [comment]
This bug is really annoying. Has there any update on this. I just reproduced it using Azure provider 3.81.0.

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v3.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
